internal/transport/keepalive_test.go

```diff
@@ -237,13 +226,10 @@ func (s) TestKeepaliveServerWithResponsiveClient(t *testing.T) {
 	}()
 
 	// Give keepalive logic some time by sleeping.
-	time.Sleep(4 * time.Second)
+	time.Sleep(500 * time.Millisecond)
 
-	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
-	defer cancel()
-	// Make sure the client transport is healthy.
-	if _, err := client.NewStream(ctx, &CallHdr{}); err != nil {
-		t.Fatalf("client.NewStream() failed: %v", err)
+	if err := pollForHealthyStream(client); err != nil {
```

Comment:
Why "poll"?  It should succeed on the first attempt always, no?  Same for all call sites?